### PR TITLE
Refactor SendMessage to be testable, also requires refactoring the runner generator

### PR DESCRIPTION
**Pull request by hunterwerlla**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

- Refactor SendMessage to be testable
- Add functional tests
- Change generate runners to work with multiple clients (needed for a few tasks)
## Checklist

- [ ] I have read the **README** document

- [ ] Local run of `npm run fullBuild` succeeds
- [ ] I have run tslint against my changes using the root tslint.yaml
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG** (CHANGELOG.md) if the change is feature related

## License

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/TaskTests/SendMessage/SendMessage-test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    delaySeconds: undefined
+}
+
+const promiseThrowsResponse = {
+    promise: function() {
+        throw new Error('no')
+    }
+}
+
+const promiseSuceeds = {
```
Comment (awschristou) [inline tests/TaskTests/SendMessage/SendMessage-test.ts]: typo

File: tests/TaskTests/SendMessage/SendMessage-test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    test('Send message to queue when message target is not topic', () => {
+        expect.assertions(1)
+        const sqs = new SQS() as any
+        sqs.getQueueAttributes = jest.fn(() => undefined)
+        const taskOperations = new TaskOperations(new SNS(), sqs, defaultTaskParameters)
+        taskOperations.execute().catch()
+        expect(sqs.getQueueAttributes).toBeCalled()
+    })
+
+    test('Send message to queue when message to topic when target is topic', () => {
```
Comment (awschristou) [inline tests/TaskTests/SendMessage/SendMessage-test.ts]: Please reword

Comment (hunterwerlla) [reply tests/TaskTests/SendMessage/SendMessage-test.ts]: This was a disaster of a sentence, reworded

File: tests/TaskTests/SendMessage/SendMessage-test.ts (hunk 1 above)
Comment (hunterwerlla) [reply tests/TaskTests/SendMessage/SendMessage-test.ts]: fixed

**Pull request merged**